File: mrshudson-backend/src/main/java/com/mrshudson/domain/dto/LoginResponse.java
package com.mrshudson.domain.dto;

import com.mrshudson.domain.entity.User;
import lombok.Data;

/**
 * 登录响应DTO
 */
@Data
public class LoginResponse {

    private Long id;
    private String username;
    private String accessToken;
    private String refreshToken;
    private Long expiresIn;

    /**
     * 从User实体创建响应（不包含Token）
     */
    public static LoginResponse fromUser(User user) {
        LoginResponse response = new LoginResponse();
        response.setId(user.getId());
        response.setUsername(user.getUsername());
        return response;
    }

    /**
     * 从User实体创建响应（包含Token）
     */
    public static LoginResponse fromUser(User user, String accessToken, String refreshToken, Long expiresIn) {
        LoginResponse response = fromUser(user);
        response.setAccessToken(accessToken);
        response.setRefreshToken(refreshToken);
        response.setExpiresIn(expiresIn);
        return response;
    }
}

File: mrshudson-android/app/src/main/java/com/mrshudson/android/data/remote/dto/LoginResponse.kt
package com.mrshudson.android.data.remote.dto

/**
 * 登录响应数据类
 * 包含登录成功后返回的 Token 信息
 *
 * @property id 用户ID
 * @property username 用户名
 * @property accessToken JWT 访问令牌，用于 API 请求认证
 * @property refreshToken JWT 刷新令牌，用于获取新的访问令牌
 * @property expiresIn 令牌过期时间（秒）
 */
data class LoginResponse(
    val id: Long,
    val username: String,
    val accessToken: String,
    val refreshToken: String,
    val expiresIn: Long? = null
)
